### PR TITLE
[OZ][L-04] Treasury Fees Can Be Locked in the SFC

### PR DESCRIPTION
**Pull request by Mike-CZ**
This PR adds collection of failed trasury fees and option to claim them later.

### CODE REVIEW

File: contracts/sfc/SFC.sol [hunk 1]
```diff
@@ -419,6 +429,22 @@ contract SFC is OwnableUpgradeable, UUPSUpgradeable, Version {
...
+        }
+        if (unresolvedTreasuryFees == 0) {
+            revert NoUnresolvedTreasuryFees();
+        }
+        (bool success, ) = treasuryAddress.call{value: unresolvedTreasuryFees, gas: 1000000}("");
+        if (!success) {
+            revert TransferFailed();
+        }
+        emit TreasuryFeesResolved(unresolvedTreasuryFees);
+        unresolvedTreasuryFees = 0;
```
Comment (thaarok) [inline contracts/sfc/SFC.sol:445]: To follow Checks-Effects-Interactions Pattern (SWC-107), following order should be used:
1. checks
2. effects - state modification (setting unresolvedTreasuryFees to zero)
3. interactions - other contract calls (sending the money)

See https://docs.soliditylang.org/en/latest/security-considerations.html#use-the-checks-effects-interactions-pattern

Comment (Mike-CZ) [reply contracts/sfc/SFC.sol:445]: It's using re-entracy guard

Comment (thaarok) [reply contracts/sfc/SFC.sol:445]: Please follow the convention to avoid unnecessary audit notes.

**Review by thaarok — APPROVED** [review]
Thanks!

**Review by thaarok — APPROVED** [review]

**Pull request merged**